Supprt getting a base torrc from env

adventureloop opened the pull request:
Support getting a base torrc from the BASETORRC environment variable.
This enables running onionperf in chutney style tor testing networks.

adventureloop commented:
I didn't intend to pull request to here. I wanted to pull request against master in our fork, but github betrayed me. 

That said this is intended for upstream as is.

torproject-git commented:
Does Chutney already call it BASETORRC or is this a new variable? Is it better to call it CHUTNEY_* or ONIONPERF_* instead?

This seems like a reasonable thing to do but probably conflicts with #3. Once we decide on a name it is probably easy enough for @ana-cc to add it there instead.

adventureloop commented:
This is a new variable. I have support for chutney to run shellcommand tests in my tree, I think exposing a more general name can make sense for an api we want software to support. I am also exposing SOCKS_PORT and COMMAND_PORT.

I don't really have a strong opinion about using a prefix though.

Ah, I saw it was different in our master. I did dev against @ana-cc's oneshot branch. Shouldn't be hard to merge once that lands.

Commit:
Supprt getting a base torrc from env

Support getting a base torrc from the BASETORRC environment variable.
This enables running onionperf in chutney style tor testing networks.
